### PR TITLE
15). Add topic files 31 to 34

### PR DESCRIPTION
**Pull request by jameshkramer**
@mairaw @BillWagner
This PR is for user story #1576037.

It adds four topic files, makes corresponding deletions from the C# and VB folders, and adds redirection entries.

cc: @tfosmark @marteeleigh 

Contributes to #4728

### CODE REVIEW

**Comment by BillWagner** [comment]
In general, this looks great @jameshkramer 

Once you address the build errors, I'll :shipit:

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -1125,7 +1125,8 @@
     },
     {
       "source_path": "docs/csharp/programming-guide/concepts/linq/linq-to-xml-axes.md",
-      "redirect_url": "/dotnet/csharp/programming-guide/concepts/linq/linq-to-xml-axes-overview"
+      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview"
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview",
```

File: .openpublishing.redirection.json [hunk 2]
```diff
@@ -1125,7 +1125,8 @@
...
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]: given that this redirect seems to exist already, I don't think you should add redirect true here.

```suggestion
```

**Comment by jameshkramer** [comment]
Closing and reopening because of build error.

**Comment by jameshkramer** [comment]
I rebased to the current linq-to-xml, and fixed the redirection issue.

**Comment by mairaw** [comment]
I still think you should apply my comment there. Given the article has been already redirected earlier with redirect false, adding redirect: true now will mess things up.

**Comment by jameshkramer** [comment]
Ok, thank you Maira, and than you for reviewing. I made the change. Unfortunately I still don't understand redirect_document_id. The contributor's guide says: "If you redirect multiple files to the same target file, you can only set redirect_document_id to true for one of them." Yet we are doing it for two redirections when we delete the C# and VB files. In fact we set it to true on both to solve a build problem! I'm not even clear on whether "true" counts a hit on the stats of the redirected or the target file. I guess it is on the redirected file but I am not sure.

**Comment by mairaw** [comment]
> Yet we are doing it for two redirections when we delete the C# and VB files.
Not that I'm aware. We're setting the C# ones to true and the VB ones to false.

What happens when you set this to true is that the topic ID flows from the previous topic but to the next one, redirecting all the old BI data to the "new" topic. If you already published a redirect (like in the example that I commented here), doing later might cause issues since the old article was already redirected and published before with a different ID. Does this help?

File: .openpublishing.redirection.json [hunk 3]
```diff
@@ -4521,6 +4542,31 @@
       "redirect_url": "/dotnet/standard/linq/work-global-namespaces",
       "redirect_document_id": false
     },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/linq-to-xml-axes.md",
+      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview",
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
      "redirect_document_id": false
```

File: .openpublishing.redirection.json [hunk 4]
```diff
@@ -4521,6 +4542,31 @@
...
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/linq-to-xml-axes-overview.md",
+      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview",
+      "redirect_document_id": true
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/serializing-to-an-xmlreader-invoking-xslt.md",
+      "redirect_url": "/dotnet/standard/linq/serialize-xmlreader-invoke-xslt",
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]: if C# version exists and is being redirected, this needs to be false 

```suggestion
      "redirect_document_id": false
```

File: .openpublishing.redirection.json [hunk 5]
```diff
@@ -4521,6 +4542,31 @@
...
     },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/linq-to-xml-axes.md",
+      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview",
+      "redirect_document_id": true
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/linq-to-xml-axes-overview.md",
+      "redirect_url": "/dotnet/standard/linq/linq-xml-axes-overview",
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
      "redirect_document_id": false
```

File: .openpublishing.redirection.json [hunk 6]
```diff
@@ -4521,6 +4542,31 @@
...
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/serializing-to-an-xmlreader-invoking-xslt.md",
+      "redirect_url": "/dotnet/standard/linq/serialize-xmlreader-invoke-xslt",
+      "redirect_document_id": true
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/serializing-to-files-textwriters-and-xmlwriters.md",
+      "redirect_url": "/dotnet/standard/linq/serialize-files-textwriters-xmlwriters",
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
      "redirect_document_id": false
```

File: .openpublishing.redirection.json [hunk 7]
```diff
@@ -4521,6 +4542,31 @@
...
+    },
+    {
+      "source_path": "docs/visual-basic/programming-guide/concepts/linq/serializing-with-an-xml-declaration.md",
+      "redirect_url": "/dotnet/standard/linq/serialize-xml-declaration",
+      "redirect_document_id": true
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
      "redirect_document_id": false
```

**Comment by jameshkramer** [comment]
Maira, I'll check and fix all the values tomorrow. Thanks for the explanation. I'll have to think about it.

**Comment by jameshkramer** [comment]
Maira, I committed your changes and I checked the past entries for values of redirect_document_id. All is well, I believe, though some of the VB entries just take the default. So I think this is all ok.

**Comment by mairaw** [comment]
Default is false, so it should be ok

**Comment by jameshkramer** [comment]
@mairaw @BillWagner 

Hi. Is this PR waiting on me?

**Comment by BillWagner** [comment]
@jameshkramer Yes. If you can fix the merge conflicts, it's ready to merge.

**Comment by jameshkramer** [comment]
@BillWagner 

Thanks, Bill. I resolved things - first time I've done that in GitHub.

**Comment by BillWagner** [comment]
Thanks @jameshkramer Merged now.

**Comment by jameshkramer** [comment]
@BillWagner 

Thanks for the merges and the kind comments. There'll be another PR today.

**Pull request merged**